candig_federation/api/federation.py

```diff
@@ -4,162 +4,293 @@
 """
 
 import requests
-
+import json
 from flask import current_app
 from requests_futures.sessions import FuturesSession
 
-from collections import Counter
+APP = current_app
 
-app = current_app
 
+class FederationResponse:
+    """
+    Class based methods utilized to store requests within instance variables
+    """
 
-class FederationResponse(object):
+    # pylint: disable=too-many-instance-attributes
+    # pylint: disable=too-many-arguments
 
-    def __init__(self, request_type, args, url,return_mimetype, request_dict):
+    def __init__(self, request, url, endpoint_path, endpoint_payload, request_dict, return_mimetype='application/json',
+                 timeout=5):
         self.results = []
         self.status = []
-        self.request = request_type
-        self.args = args
-        self.endpoint_path = args["endpoint_path"]
-        self.endpoint_payload = args["endpoint_payload"]
+        self.request = request
         self.url = url
+        self.endpoint_path = endpoint_path
+        self.endpoint_payload = endpoint_payload
         self.return_mimetype = return_mimetype
         self.request_dict = request_dict
-        self.token = "blank"
+        self.token = self.request_dict.headers['Authorization']
+
+        self.logger = APP.logger
+
+        self.header = {
+            'Content-Type': self.return_mimetype,
+            'Accept': self.return_mimetype,
+            'Federation': 'false',
+            'Authorization': self.token,
```

Comment:
I assume the token itself is already a ‘Bearer ' token?

Reply:
I've been adding the Bearer portion in my test code, but I guess I can modify things to add it here instead. Which do you think is a better way to do things?

Reply:
Actually nevermind, the bearer portion needs to be prior since Tyk needs it anyway

Reply:
Oh ok, so this won't be an issue then!